torch/csrc/generic/Tensor.cpp

```diff
@@ -32,6 +37,31 @@
   }                                                       \
 }
 
+#define COPY_FROM_ARRAY_CPU_HALF(ELTYPE, ARRAY, STORAGE, SIZE)  \
+{ \
+  ELTYPE *arrdata = (ELTYPE*)PyArray_DATA(ARRAY);               \
+  THHalf *data = STORAGE->data;                                 \
+  for (size_t i=0; i<SIZE; i++) {                               \
+    float tmp = arrdata[i];                                     \
+    data[i] = TH_float2half(tmp);                               \
+  }                                                             \
+}
+
+#define COPY_FROM_HALF_ARRAY_CPU(ELTYPE, ARRAY, STORAGE, SIZE)  \
+{ \
+  ELTYPE *arrdata = (ELTYPE*)PyArray_DATA(ARRAY);               \
+  real *data = STORAGE->data;                                   \
+  for (size_t i=0; i<SIZE; i++) {                               \
+    data[i] = TH_half2float(arrdata[i]);                        \
+  }                                                             \
+}
+
+#define COPY_FROM_HALF_ARRAY_CPU_HALF(ELTYPE, ARRAY, STORAGE, SIZE) \
+{ \
+  char *arrdata = (char*)PyArray_DATA(ARRAY);                       \
+  memcpy(STORAGE->data, arrdata, SIZE * sizeof(float) / 2);         \
```

Comment:
you can just hardcode the size to be `SIZE * 2`. It's guaranteed that half has exactly 2 bytes